### PR TITLE
PIN screen logic Update 

### PR DESCRIPTION
**Pull request by vineyugave**
Updated PR based on some review here : 
https://github.com/Glucosio/glucosio-android/pull/319

Had to delete and create a new fork and a new PR as there were some merge conflicts that I could not solve in the original fork

Most of the changes made according to the feedback. 

The only thing **remainig** : 
Logic to **SET** Pin. 

Need feedback as to where do we want to do this. Should we just add another field to the Hello Activity ? 

FYI @emartynov

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -462,7 +462,8 @@
     <string name="reminder_item_title_generic">It is time to do an entry in Glucosio!</string>
     <string name="reminder_item_title_generic2">Time to log your glucose level!</string>
     <string name="warning_internet_connection_required">Sorry but you need internet for this feature</string>
-
+    <string name="PIN_Message">Please enter your PIN below ...</string>
```
Comment (satya164) [inline app/src/main/res/values/strings.xml]: `Please enter your PIN` and remove the ` ...`

File: app/src/main/java/org/glucosio/android/tools/TimeElapsed.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    public static void setStartTime() {
+        TimeElapsed.startTime = System.currentTimeMillis();
+    }
+
+    public static void resetTime(){
+        TimeElapsed.startTime=0;
+        TimeElapsed.endTime=0;
+    }
+    public static long getEndTime() {
```
Comment (satya164) [inline app/src/main/java/org/glucosio/android/tools/TimeElapsed.java]: nit: new line

File: app/src/main/java/org/glucosio/android/tools/TimeElapsed.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public static long getStartTime() {
+        return startTime;
+    }
+
+    public static void setStartTime() {
+        TimeElapsed.startTime = System.currentTimeMillis();
+    }
+
+    public static void resetTime(){
+        TimeElapsed.startTime=0;
```
Comment (satya164) [inline app/src/main/java/org/glucosio/android/tools/TimeElapsed.java]: nit: space around `=`

File: app/src/main/java/org/glucosio/android/tools/TimeElapsed.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    public static long getStartTime() {
+        return startTime;
+    }
+
+    public static void setStartTime() {
+        TimeElapsed.startTime = System.currentTimeMillis();
+    }
+
+    public static void resetTime(){
```
Comment (satya164) [inline app/src/main/java/org/glucosio/android/tools/TimeElapsed.java]: nit: space after `()`

File: app/src/main/java/org/glucosio/android/tools/TimeElapsed.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    private static TimeElapsed _instance = null;
+    private static long startTime;
+    private static long endTime;
+
+    protected TimeElapsed() {
+        // Exists only to defeat instantiation.
+    }
+
+    public static TimeElapsed getInstance(){
+        if(_instance == null){
```
Comment (satya164) [inline app/src/main/java/org/glucosio/android/tools/TimeElapsed.java]: nit: space around `(...)`

File: app/src/main/java/org/glucosio/android/tools/TimeElapsed.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    private static TimeElapsed _instance = null;
+    private static long startTime;
+    private static long endTime;
+
+    protected TimeElapsed() {
+        // Exists only to defeat instantiation.
+    }
+
+    public static TimeElapsed getInstance(){
```
Comment (satya164) [inline app/src/main/java/org/glucosio/android/tools/TimeElapsed.java]: nit: space after `)`

File: app/src/main/java/org/glucosio/android/activity/ParentActivity.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
+package org.glucosio.android.activity;
```
Comment (satya164) [inline app/src/main/java/org/glucosio/android/activity/ParentActivity.java]: I think `ParentActivity` is not a very good name

File: app/src/main/java/org/glucosio/android/activity/ParentActivity.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        //check if app has passed a time threshold
+        if(timeElapsedInstance.getStartTime() != 0){
+            timeElapsedInstance.setEndTime(Calendar.getInstance().getTimeInMillis());
+            long threshold = timeElapsedInstance.getEndTime()-timeElapsedInstance.getStartTime();
+            Log.d(TAG,"Threshold : "+threshold);
+            //Current timeout threshold set to 30s
+            if(threshold>30000){
+                Intent intent = new Intent(this, AppLockActivity.class);
+                startActivity(intent);
+            }else{
```
Comment (satya164) [inline app/src/main/java/org/glucosio/android/activity/ParentActivity.java]: nit: remove unnecessary `else`

File: app/src/main/java/org/glucosio/android/activity/ParentActivity.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        super.onCreate(savedInstanceState);
+        timeElapsedInstance = TimeElapsed.getInstance();
+        timeElapsedInstance.resetTime();
+        timeElapsedInstance.setStartTime();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        //check if app has passed a time threshold
```
Comment (satya164) [inline app/src/main/java/org/glucosio/android/activity/ParentActivity.java]: nit: spaces around parenthesis in the whole method

**Comment by bkerensa** [comment]
We want to set this in settings and enable it in settings not HelloActivity and it should be off by default.

We will need a "Security" settings submenu and we the pin screen when enabled should allow pin or fingerprint or both.

@satya164 We will need a mockup for this ideally Glucosio pink pin screen with our rounded Glucosio logo

**Comment by vineyugave** [comment]
@bkerensa gotcha ... I think that makes sense.. do you know who can do the mockups ? I am not a designer.. If they could give it to me, I can start putting it together.. thanks

**Comment by satya164** [comment]
image

**Comment by emartynov** [comment]
Any updates on this?

**Comment by vineyugave** [comment]
I have been really busy with work ... Will try to finish this in 2-3 weeks as I get time ...sorry for the delay

**Comment by emartynov** [comment]
Closing as for now. Please reopen when you resume work.

**Pull request closed without merging**